interfaces/acados_matlab_octave/AcadosOcpSolver.m

```diff
@@ -112,22 +112,28 @@ function set(obj, field, value, varargin)
 
                 if length(varargin) > 0
                     n = varargin{1};
-                    Q = obj.get('qp_Q', n);
-                    R = obj.get('qp_R', n);
-                    S = obj.get('qp_S', n);
 
-                    value = [R, S; S', Q];
+                    if n < obj.ocp.solver_options.N_horizon
+                        Q = obj.get('qp_Q', n);
+                        R = obj.get('qp_R', n);
+                        S = obj.get('qp_S', n);
+
+                        value = [R, S; S', Q];
+                    else
+                        value = obj.get('qp_Q', n);
```

Comment:
What happens, if someone requests `'qp_R'` on the last stage?

Reply:
this is what I changed with the additional if here: `if n < obj.ocp.solver_options.N_horizon`